---
title: "1_LA_QC_Steps"
author: "Peter Fiorica"
date: "September 11, 2019"
output: html_document
---

# Introduction
This file contains the notes for the genome-wide genotype quality control of the Latin American Cohort of our multi-ethnic transcriptome-wide association study of prostate cancer.  These notes and protocol are nearly identical to those of the [QC Steps for the African American and Japanese American Populations](https://github.com/WheelerLab/Prostate-Cancer-Study).  Since I have already made two sets of QC notes already (one which is titled after Star Wars Films and another that is cut and dry), I think I will be refering to the QC steps here with baseball allegories.  With that, standing at 2147 subjects (1081 cases & 1066 controls) and 657366 SNPs, taking the mound for the LA QC will be `/home/peter/jl_prostate_cancer/genotypes_dbGaP\la_starting_genotypes`.

#Quality Control
## Sex and Heterozygous Haploid Check
Leading off for the LA QC is the left fielder, sex and .hh check
```
plink --bfile /home/peter/jl_prostate_cancer/genotypes_dbGaP/la_starting_genotypes --set-hh-missing --make-bed --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/1_la_procan_hh_setmissing

plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/1_la_procan_hh_setmissing --check-sex --missing --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/2_la_procan_sexchecked
```

## Filtering by Genotyping Rate
### Identifying the unfiltered genotyping rate
Hitting second for the LA QC is the second baseman, unfiltered genotyping rate
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/1_la_procan_hh_setmissing --missing --out //home/peter/jl_prostate_cancer/qc_steps/la_qc/3_la_procan_hh_setmissing
```

```{r, warning=FALSE, cache=TRUE}
library(dplyr)
library(data.table)
library(ggplot2)
"%&%"=function(a,b) paste(a,b,sep="")
my.dir<-"Z://jl_prostate_cancer/qc_steps/la_qc/"

lmiss<-fread(my.dir%&%"3_la_procan_hh_setmissing.lmiss", header = T)
hist(lmiss$F_MISS)
dim(lmiss)[1]
#This is the number of SNPs we have before filtering
table(lmiss$F_MISS<0.01)
#The TRUE value is how many SNPs have a genotyping rate > 99%.
sum(lmiss$F_MISS<0.01)/(dim(lmiss)[1])
```

### Filtering by genotyping rate & identifying filtered genotyping rate
In the three hole, the shortstop, filtered genotyping rate
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/1_la_procan_hh_setmissing --geno 0.01 --make-bed --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/4_la_procan_filtered

plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/4_la_procan_filtered --missing --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/5_la_procan_filtered_missingess
```
```{r, cache = TRUE}
newimiss<-fread(my.dir%&%"5_la_procan_filtered_missingess.imiss")
hist(newimiss$F_MISS)

newlmiss<-fread(my.dir%&%"5_la_procan_filtered_missingess.lmiss")
hist(newlmiss$F_MISS)
dim(newlmiss)[1]
```

## Filtering by Hardy-Weinberg Equilibrium
Hitting cleanup: First baseman: Hardy-Weinberg Equlibrium Filtering.  I like to think that G.H. Hardy and Wilhelm Weinberg were decent baseball players. Hopefully one of them could fill the four spot in a batting order.  Anyway, we're removing SNPs outside of HWE ( $P<1^{-6}$).
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/4_la_procan_filtered --hardy --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/6_la_procan_hardy
```

```{r, cache = TRUE}
hwe<-fread(my.dir%&%"6_la_procan_hardy.hwe",header =T)
summary(hwe$P)

hist(hwe$P)
abline(v = median(hwe$P)+sd(hwe$P), col = "blue", lwd = 2)
abline(v = median(hwe$P)-sd(hwe$P), col = "blue", lwd = 2)
abline(v = median(hwe$P)+1.5*sd(hwe$P), col = "red")
abline(v = median(hwe$P)-1.5*sd(hwe$P), col = "red")

table(hwe$P<1e-6)

table(hwe$P<1e-06)/sum(table(hwe$P<1e-06))

outlierSNPs<-as.data.table(subset(hwe$SNP,hwe$P<=1e-6))
fwrite(outlierSNPs, my.dir%&%"la_HWEoutlierSNPstoberemoved.txt", col.names = F, row.names = F, sep = " ", quote = F)
```
Alright, so it looks like there are 1456 to be removed because they deviate from HWE significantly.

```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/4_la_procan_filtered --exclude la_HWEoutlierSNPstoberemoved.txt --make-bed --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/7_la_procan_removed_hwe_outliers
```

## Identity by Descent Pruning
### Determine IBD Values
At the heart of the order in the five spot, the designated hitter, IBD pruning.
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/7_la_procan_removed_hwe_outliers --indep-pairwise 50 5 0.3 --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/8_la_procan_ibd_values
```

### Extract SNPs with excess IBD
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/7_la_procan_removed_hwe_outliers --extract /home/peter/jl_prostate_cancer/qc_steps/la_qc/8_la_procan_ibd_values.prune.in --genome --min 0.025 --out /home/peter/jl_prostate_cancer/qc_steps/la_qc/9_la_procan_ibd_extracted_snps
```

```{r, cache = TRUE}
ibd<-fread(my.dir %&% "9_la_procan_ibd_extracted_snps.genome",header=T)

ggplot(data = ibd, aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()
#Here is the initial data in which each point represents a pair of individuals to show relatedness.
#The plot below is identifcal to the plot above with the exception that individuals with a PI HAT>0.2 as red points.
ggplot(data = ibd, aes(x=Z0,y=Z1))+
    geom_point(alpha=1/4, color = ifelse(ibd$PI_HAT>0.2,'red','black'))+theme_bw()
```


[Turner et al](doi:10.1002/0471142905.hg0119s68), 2011 provide an overview of the GWAS QC process.  Notably, they provide a great overview of what each point represents at each Z1 and Z2 value
![Turner, S. , Armstrong, L. L., Bradford, Y. , Carlson, C. S., Crawford, D. C., Crenshaw, A. T., Andrade, M. , Doheny, K. F., Haines, J. L., Hayes, G. , Jarvik, G. , Jiang, L. , Kullo, I. J., Li, R. , Ling, H. , Manolio, T. A., Matsumoto, M. , McCarty, C. A., McDavid, A. N., Mirel, D. B., Paschall, J. E., Pugh, E. W., Rasmussen, L. V., Wilke, R. A., Zuvich, R. L. and Ritchie, M. D. (2011), Quality Control Procedures for Genome-Wide Association Studies. Current Protocols in Human Genetics, 68: 1.19.1-1.19.18.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3066182/bin/nihms-262648-f0004.jpg)

```{r,cache = TRUE}
dups<-data.frame()
for( i in 1:dim(ibd)[1]){
  if(as.character(ibd$IID1[i])==as.character(ibd$IID2[i])){
    dups<-rbind(dups,ibd[i,])
  }
}
dim(dups)
```

Next, we will check for HapMap individuals, but I do not expect there to be any.
```{r,cache = TRUE}
hapmap <- filter(ibd,grepl('NA',IID1))
#No hapmap individuals. No surpise.
toExclude <- c(as.character(dups$IID1),as.character(hapmap$IID1))
a <- as.character(ibd$IID1) %in% toExclude
others <- ibd[a==FALSE,]
#Isolating individuals that need to be removed.
toremove<-filter(others,PI_HAT>=0.2)
write.table(toremove,my.dir%&%"la_procan_Relate.to.remove.txt",quote=FALSE, row.names = FALSE)
```
We need to flag 41 individuals from our study.  We have already agreed that we plan to use GEMMA for PrediXcan and relatedness, so we can increase our power slightly.

## Identifying individuals with excess heterozygosity
Hitting sixth is the third baseman, excess heterozygosity filtering
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/7_la_procan_removed_hwe_outliers --extract 8_la_procan_ibd_values.prune.in --het --out 10_la_procan_heterozygosity_estimates
```
```{r,cache = TRUE}
HET<-fread(my.dir%&%"10_la_procan_heterozygosity_estimates.het",header =T)
h=HET$"N(NM)"-HET$"O(HOM)"/HET$"N(NM)"
oldpar = par(mfrow=c(1,2))
hist(h,50)

summary(HET$F)

hist(HET$F,50)
abline(v=mean(HET$F)+6*sd(HET$F),col="red")
abline(v=mean(HET$F)-6*sd(HET$F),col="red")

sortHET <- HET[order(HET$F),]
outliers <- data.table()

for(i in 1:length(sortHET$F)){
  if(sortHET[i,6] > (mean(sortHET$F)+3*sd(sortHET$F))){
    outliers <- rbind(outliers, sortHET[i,])
  }
  if(sortHET[i,6] < (mean(sortHET$F)-3*sd(sortHET$F))){
    outliers <- rbind(outliers, sortHET[i,])
  }
}

hetoutliers <- select(outliers, FID, IID)
dim(hetoutliers)
#It looks like we are working with 38 heterozygosity outliers.
fwrite(hetoutliers, "Z://jl_prostate_cancer/qc_steps/la_qc/la_procan_het_outliers.txt", quote = F, col.names = F, row.names = F, sep = " ")
```

### Removing individuals with excess heterozygosity

```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/7_la_procan_removed_hwe_outliers --remove la_procan_het_outliers.txt --extract 8_la_procan_ibd_values.prune.in --make-bed --out 10_la_procan_removed_het_outliers

plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/10_la_procan_removed_het_outliers --remove la_procan_Relate.to.remove.txt --extract 8_la_procan_ibd_values.prune.in --genome --min 0.025 --out 11_la_procan_removed_het_and_related_forIBD
```
### Plotting IBD after removing individuals with excess heterozygosity and relatedness
```{r, cache = TRUE}
IBD<-fread(my.dir %&% "11_la_procan_removed_het_and_related_forIBD.genome" ,header=T)
ggplot(data = IBD, aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()

ggplot(data = IBD, aes(x=Z0,y=Z1))+geom_point(alpha=1/4, color = ifelse(IBD$PI_HAT>0.2,'red','black'))+theme_bw()
#The plots are the same. Great! That means we removed all of the individuals previously marked as related.
```

```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/10_la_procan_removed_het_outliers --extract 8_la_procan_ibd_values.prune.in --remove la_procan_Relate.to.remove.txt --make-bed --out 12_la_procan_removed_related
```
## Principal Component Analysis
Hitting in the seven spot is the right fielder, principal component analysis
As a heads up we are merging with a set of HapMap phase three genotypes from individuals of YRI (Nigeria), CEU (Utah), and ASN (Japan & Beijing) to confirm self identified ancestry.
### Failed first merge attempt
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/12_la_procan_removed_related --bmerge /home/wheelerlab1/Data/HAPMAP3_hg18/HM3_ASN_CEU_YRI_Unrelated_hg18_noAmbig --make-bed --out 13_la_procan_failed_hapmap_merge
```

### Removing SNPs with more than three alleles present
Batting eighth, playing catcher is removing SNPS with +3 alleles
```
plink --bfile /home/wheelerlab1/Data/HAPMAP3_hg18/HM3_ASN_CEU_YRI_Unrelated_hg18_noAmbig --exclude /home/peter/jl_prostate_cancer/qc_steps/la_qc/13_la_procan_failed_hapmap_merge-merge.missnp --make-bed --out 14_la_procan_hapmap_files_for_merge
```

### Merge attempt with HapMap number two
```
plink --bfile /home/peter/jl_prostate_cancer/qc_steps/la_qc/12_la_procan_removed_related --bmerge 14_la_procan_hapmap_files_for_merge --make-bed --out 15_la_procan_merged_w_hapmap
```

### Run PCA in PLINK
```
plink --bfile 15_la_procan_merged_w_hapmap --geno 0.01 --maf 0.05 --chr 1-22 --pca 10 --out 16_la_procan_hapmap_gwas_pca
```

## Plot PCA Values
Rounding out the order is the center fielder, plotting the PC values and making a scree plot.
```{r, cache = TRUE}
hapmappopinfo<-read.table("Z://prostate_cancer/QC_Steps/step6/pop_HM3_hg18_forPCA.txt")%>%select(V1,V3)
colnames(hapmappopinfo) <- c("pop","IID")

fam <-fread(my.dir%&%"15_la_procan_merged_w_hapmap.fam", header =F)%>%select(V1,V2)
colnames(fam) <- c("FID","IID")

popinfo <- left_join(fam,hapmappopinfo,by="IID")

popinfo <-mutate(popinfo, pop=ifelse(is.na(pop),'GWAS',as.character(pop)))
table(popinfo$pop)

pcs<-read.table(my.dir%&%"16_la_procan_hapmap_gwas_pca.eigenvec")
pcdf <- data.frame(popinfo, pcs[,3:12]) %>% rename (PC1=V3,PC2=V4,PC3=V5,PC4=V6,PC5=V7,PC6=V8,PC7=V9,PC8=V10,PC9=V11,PC10=V12)
gwas <- filter(pcdf,pop=='GWAS')
hm3 <- filter(pcdf, grepl('NA',IID))
eval<-scan(my.dir%&%"16_la_procan_hapmap_gwas_pca.eigenval")[1:10]
pve<-eval/sum(eval)#Calculate the percent explained by each PC
PCs<-c(1:10)
PVE<-data.table(PCs,pve)

ggplot() + geom_point(data=gwas,aes(x=PC1,y=PC2,col=pop,shape=pop))+geom_point(data=hm3,aes(x=PC1,y=PC2,col=pop,shape=pop))+ theme_bw() + scale_colour_brewer(palette="Set1")

ggplot() + geom_point(data=gwas,aes(x=PC1,y=PC3,col=pop,shape=pop))+geom_point(data=hm3,aes(x=PC1,y=PC2,col=pop,shape=pop))+ theme_bw() + scale_colour_brewer(palette="Set1")

ggplot() + geom_point(data=gwas,aes(x=PC2,y=PC3,col=pop,shape=pop))+geom_point(data=hm3,aes(x=PC1,y=PC2,col=pop,shape=pop))+ theme_bw() + scale_colour_brewer(palette="Set1")

ggplot(data = PVE, aes(y=pve, x = PCs)) + geom_point() + geom_line() + xlab("PC") + ylab("PVE") + ggtitle("Scree Plot Prostate Cancer") + theme_bw()
```

Each of the three PCA plots helps elucidate the ancestry of the data. Since we have previously been working with African American and Japanese populations, these PCA plots may seem unfamiliar.  However, they make sense since individuals of Latin American descent will have their genotypes made of of chromosome blocks from Europeans in the Iberian peninsula, and indigenous populations of Central America.

# Next Steps
## Liftover
Right now, the data is in genome build hg18. We need to lift it over to hg19. A good example of the liftover process can be found at https://github.com/WheelerLab/Prostate-Cancer-Study/blob/master/AA_Prostate_Cancer/liftover_and_preimputation.Rmd. When we perform the liftover, we will use `/home/peter/jl_prostate_cancer/qc_steps/7_jl_procan_removed_hwe_outliers` since this set of files includes unpruned data with HWE outliers removed.

## Imputation
After we convert the genotypes to genome build 19, we will upload the data to the [University of Michigan Imputation Server](https://imputationserver.sph.umich.edu/index.html#!).  Following imputation we will filter the data by $R^{2}>0.8$ and $MAF > 0.01$.