---
title: "DAP-G Output"
author: "Peter Fiorica"
date: "April 3, 2020"
output: html_document
---
```{r}
library(susieR)
library(dplyr)
library(ggplot2)
library(data.table)
library(Matrix)
library(ggrepel)
library(stringr)
```

# African American Prostate Cancer Data
```{r}
dapg_snps<-fread("Z://dapg_stuff/pro_can_chr8_dapg_SNP_summary.txt", header = F)
dapg_model<-fread("Z://dapg_stuff/pro_can_chr8_dapg_model_summary.txt", header = F)
dapg_cluster<-fread("Z://dapg_stuff/pro_can_chr8_dapg_clusters.txt", header = F)

#gwas<-fread("Z://prostate_cancer/locuszoom_chr8.assoc.logistic", header = T)
gwas_chr8<-fread('Z://prostate_cancer/chromosome_8.assoc.logistic', header = F )

#zscore object is generated by running the susier script.
gwasnzscore<-left_join(gwas_chr8,zscore, by = c("V2"="SNP"))
gwasnzscore$snpid<-paste(gwasnzscore$V1,gwasnzscore$V3, sep = "_")

dapg_snps$V2<-str_sub(dapg_snps$V2, 1, 11)



#dapg_gwas<-left_join(dapg_snps, gwas_chr8, by =c("V2")

mixed<-left_join(dapg_snps, gwasnzscore, by = c("V2"="snpid"))

forplot<-dplyr::select(mixed,V1, V2, V3, V4, as.factor(mixed$V5), V6, CHR, SNP, BP,P, PIP)


#PIP Compare Plot
ggplot(forplot, aes(x=V3, y = PIP))+
  geom_point(alpha= 0.5, size = 1.5, aes(color = as.factor(V5)))+
  xlab("DAP-G PIP")+
  ylab("SuSIE PIP")+
  theme_bw(20) 

#SusieR-esque Plot
ggplot(mixed, aes(x=V3.y, y = V3.x))+
  geom_point(alpha= 0.5, size = 2, aes(color = as.factor(V5.x)))+
  xlab("Chromosome 8 Location")+
  ylab("DAP-G PIP")+
  ggtitle("African American DAPG")+
  theme_bw(20)

```

## PIP V. P-Value Plot
```{r}

gwas_chr8<-fread("Z://prostate_cancer/chromosome_8.assoc.logistic", header = F)

gwas_chr8$snpid<-paste(gwas_chr8$V1,gwas_chr8$V3, sep = "_")

mixed8<-left_join(dapg_snps,gwas_chr8, by = c("V2"= "snpid"))

p<-ggplot(mixed8, aes(x= -log10(mixed8$V9), y = mixed8$V3.x))+
  geom_point(size = 4, alpha = 0.2, aes(color = as.factor(mixed8$V5.x)))+
  geom_vline(xintercept= -log10(5e-8), color = "red", alpha = 0.2, size = 2)+
  xlab("-log(GWAS P-Value)")+
  ylab("DAP-G PIP")+
  guides(fill= guide_legend(name = "DAP-G Cluser")) +
  theme_bw(18)

```


# Japanese American Prostate Cancer Data
```{r}
dapg_snps<-fread("Z://dapg_stuff/ja_pro_can_chr8_dapg_SNP_summary.txt", header = F)
dapg_model<-fread("Z://dapg_stuff/ja_pro_can_chr8_dapg_model_summary.txt", header = F)
dapg_cluster<-fread("Z://dapg_stuff/ja_pro_can_chr8_dapg_cluster_summary.txt", header = F)

ja_results<-fread("Z://prostate_cancer/jl_prostate_cancer/gwas/ja_chromosome_results/8.txt", header = F)

gwasnzscore<-left_join(ja_results,zscore, by = c("V2"="SNP"))
gwasnzscore$snpid<-paste(gwasnzscore$V1,gwasnzscore$V3, sep = "_")
gwasnzscore<-gwasnzscore[complete.cases(gwasnzscore),]

dapg_snps$V2<-str_sub(dapg_snps$V2, 1, 11)


mixed<-left_join(dapg_snps, gwasnzscore, by = c("V2"="snpid"))

#forplot<-dplyr::select(mixed,V1, V2, V3, V4, as.factor(mixed$V5))

#Susier vs. DAP-G Plot
ggplot(mixed, aes(x=V3.x, y = PIP))+
  geom_point(alpha= 0.5, size = 2, aes(color = as.factor(V5.x)))+
  xlab("DAP-G PIP")+
  ylab("SuSIE PIP")+
  ggtitle("Japanese American Chromosome 8 PIP Compare")+
  theme_bw(20) 


#PIP Log(P) Plot
ggplot(mixed, aes(y=V3.x, x = -log10(V9)))+
  geom_point(alpha= 0.5, size = 1.5, aes(color = as.factor(V5.x), size = V3.x))+
 geom_vline(xintercept= -log10(5e-8), color = "red", alpha = 0.2, size = 2)+
  xlab("-log(GWAS P-Value)")+
  ylab("DAP-G PIP")+
  guides(fill= guide_legend(name = "DAP-G Cluser")) +
  ggtitle("Japanese American Prostate Cancer (CHR 8)")+
  theme_bw(18)

ggplot(mixed, aes(x=V3.y, y = V3.x))+
  geom_point(alpha= 0.5, size = 2, aes(color = as.factor(V5.x)))+
  xlab("Chromosome 8 Location")+
  ylab("DAP-G PIP")+
  ggtitle("Japanese American DAPG")+
  theme_bw(20)
```